server/controllers/login.js

```diff
@@ -0,0 +1,48 @@
+const bcrypt = require("bcrypt");
+const { sign } = require("jsonwebtoken");
+const getPassword = require("../database/queries/getPassword");
+require("dotenv").config();
+const { SECRET } = process.env;
+const createToken = (email, secret) => {
+  return sign({ email }, secret);
+};
+
+exports.login = (req, res) => {
+  const { email, password } = req.body;
+  if (!email || !password) {
```

Comment:
The check should not be in BE, you should make sure receive non-empty values

Reply:
actually that's not correct, in fact validations always have to be done in the backend since that's where the logic is executed and it's trivial to bypass validation on the frontend - an attacker can simply create an HTTP request himself using Postman or SoapUI, completely avoiding any interaction with the FE. The frontend validation is in fact just a convenience for the users so their mistakes are discovered early (so there is no need to send a request). But you can't rely on it as a real validation